### DISCUSSION TITLE
Events not working in NS2

### DISCUSSION THREAD

**Discussion by dlinsen1** [petersenpeter/CellExplorer#103] [Q&A] [answered]
For some reason my events are not being detected by NS2 and even when i generated a new events file within NS2 it is not detected or displayed. events work as usual in NS1. any help is much appreciated.

**Comment by petersenpeter** [comment]
Hi 

I need some more details on this issue to help. Could you share your *.events.mat file and I can try to troubleshoot. 

When you say you generate an events file within NS2, is this from the Analysis menu (Analysis->Events->Create_empty_events_fie)? Do you see an error in Matlab?

**Comment by dlinsen1** [comment]
Your questions sparked some ideas that allowed me to resolve the strange issue, which I am still unsure the cause of. Basically i had an events file generated from the standalone function bz_findripples that would not autoload into NS2. I then made a new file from analysis menu like you suggested and pasted in timestamps - still didnt load. i finally deleted it, reran bz_ripples from the NS2 GUI and everything loaded up as it should.

**Comment by petersenpeter** [comment, marked as answer]
As with your metadata, I believe this is also caused by deviation from the events format. You can find an example [rippe.events.mat file here](https://buzsakilab.nyumc.org/datasets/PetersenP/CellExplorerExampleData/MS22/Peter_MS22_180629_110319_concat/Peter_MS22_180629_110319_concat.ripples.events.mat), and documentation [here](https://cellexplorer.org/datastructure/data-structure-and-format/#events). 

The struct has to have the same name (StructName) as the name in the filename. basename.StructName.events.mat, and the struct has to have a timestamps field with correct dimensions.
